src/main/java/com/cloudbees/jenkins/plugins/BitbucketJobProbe.java

```diff
@@ -112,6 +107,24 @@ private boolean match(SCM scm, URIish url) {
         return false;
     }
 
+    public boolean looselyMatch(URIish url, URIish urIish) {
+        LOGGER.log(Level.FINE, "repo scheme: {0}", urIish.getScheme());
+        // needed cause the SSH and HTTPS URI differs in Bitbucket Server.
+        if ("ssh".equalsIgnoreCase(urIish.getScheme().toLowerCase())) {
+            // SSH repositories structure: ssh://git@domain:port/<project>/<repo>
+            // so strip everything in the url before /<project>/<repo
+            int indexOfLastSlash = url.getPath().lastIndexOf('/');
+            int indexOfLastButOneSlash = url.getPath().lastIndexOf('/', indexOfLastSlash - 1);
+            String projectRepoPath = url.getPath().substring(indexOfLastButOneSlash);
+            url = url.setPath(projectRepoPath);
+        } else /* assume HTTP(S) */ {
+            // HTTP clone repository contains '/scm' somewhere after the context path.
+            urIish = urIish.setPath(urIish.getPath().replaceFirst("/scm", ""));
```

Comment:
How about using ("/scm/", "/") instead just in case someone uses a really unfortunate name there. For example http://example.org/scmtool/scm/projects/prj1/repos/repo1